engine.js

```diff
@@ -331,7 +397,65 @@ var hmdajson = require('./lib/hmdajson'),
      * -----------------------------------------------------
      */
 
-    // TODO
+    var bindArg = function(arg, objList) {
+        var tokens = arg.split('.');
+        for (var i = 0; i < objList.length; i++) {
+            var mappedArg = objList[i];
+
+            for (var j = 0; j < tokens.length; j++) {
+                mappedArg = mappedArg[tokens[j]];
+                if (mappedArg === undefined) {
+                    break;
+                }
+            }
+            
+            if (mappedArg !== undefined) {
+                return mappedArg;
+            }
+        }
+        return arg;
+    }; 
+
+    HMDAEngine.execRule = function(topLevelObj, rule) {
+        var result = {
+            argIndex: 0,
+            args: [],
+            body: '',
+            properties: {}
+        };
+
+        HMDAEngine.parseRule(rule, result);
+        result.body = 'return ' + result.body + ';';
+
+        var args = _.map(result.args, function(arg) {
+            if (typeof(arg) === 'string') {
+                var objList = [topLevelObj, root];        // Context list to search
```

Comment:
If you're naming this `objList` and then commenting that it is the context list, you should probably just name it `contextList` or `contexts`.